scripts/connector-version-radar.py

```
@@ -610,28 +612,159 @@ def mark_state(
+    if metadata.st_size > MAX_LIVE_EVIDENCE_BYTES:
+        raise RadarError(f"{label} exceeds {MAX_LIVE_EVIDENCE_BYTES} bytes: {path}")
+    try:
+        raw = path.read_bytes()
+    except OSError as exc:
+        raise RadarError(f"could not read {label} {path}: {exc}") from exc
+    if len(raw) > MAX_LIVE_EVIDENCE_BYTES:
+        raise RadarError(f"{label} exceeds {MAX_LIVE_EVIDENCE_BYTES} bytes: {path}")
+    try:
+        return json.loads(raw)
+    except (UnicodeDecodeError, json.JSONDecodeError) as exc:
+        raise RadarError(f"{label} is not valid UTF-8 JSON: {path}: {exc}") from exc
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)